### PR TITLE
refactor(udf): switch to the latest arrow-udf versions

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Currently, decimal and jsonb types are mapped to Arrow LargeBinary and LargeString types in UDFs. However, since we have decoupled UDF code into the `arrow-udf` project, this design blocks community users from using LargeBinary and LargeString types in Arrow. (https://github.com/risingwavelabs/arrow-udf/issues/16) To solve this problem, we decided to use [Arrow extension types](https://github.com/risingwavelabs/arrow-udf?tab=readme-ov-file#extension-types) for these special types. The migration on the arrow-udf side is completed. (https://github.com/risingwavelabs/arrow-udf/pull/17, https://github.com/risingwavelabs/arrow-udf/pull/18) This PR finishes the rest work on the RisingWave side, which is, switching to new arrow-udf versions and handling backward compatibility issues.

In detail, this PR:

- removes `risingwave-udf` crate. the code is moved to [arrow-udf-flight](https://github.com/risingwavelabs/arrow-udf/tree/main/arrow-udf-flight).
- removes Python UDF SDK. the code is moved to [arrow-udf-flight/python](https://github.com/risingwavelabs/arrow-udf/tree/main/arrow-udf-flight/python).
- removes Java UDF SDK. the code is moved to [arrow-udf-flight/java](https://github.com/risingwavelabs/arrow-udf/tree/main/arrow-udf-flight/java).
- updates `arrow-udf-*` versions
- introduces new Arrow conversion for decimal and jsonb types and keeps the old way as legacy. (see `arrow_udf.rs`)
- checks the version of Python/Java/Rust UDF SDK at runtime to determine whether using the legacy conversion. 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

The Python UDF SDK is updated. If using RisingWave 1.10, you are encouraged to update to the latest version following the migration guide (to be added later). Older versions are still supported, but may be deprecated in the future version.

### CODE REVIEW

File: e2e_test/udf/test.py [hunk 1]
```diff
@@ -84,7 +90,7 @@ def hex_to_dec(hex: Optional[str]) -> Optional[Decimal]:
     return dec
 
 
-@udf(input_types=["FLOAT8"], result_type="DECIMAL")
+@udf(input_types=["FLOAT64"], result_type="DECIMAL")
```
Comment (BugenZhao) [inline e2e_test/udf/test.py:93]: Is this considered a breaking change that should be notable for existing users?

File: src/expr/core/src/table_function/user_defined.rs [hunk 2]
```diff
@@ -203,12 +189,18 @@ pub fn new_user_defined(prost: &PbTableFunction, chunk_size: usize) -> Result<Bo
...
 
+    let mut arrow_convert = UdfArrowConvert::default();
+
     let client = match udtf.language.as_str() {
         "wasm" | "rust" => {
             let compressed_wasm_binary = udtf.get_compressed_binary()?;
             let wasm_binary = zstd::stream::decode_all(compressed_wasm_binary.as_slice())
                 .context("failed to decompress wasm binary")?;
             let runtime = crate::expr::expr_udf::get_or_create_wasm_runtime(&wasm_binary)?;
+            // backward compatibility
```
Comment (BugenZhao) [inline src/expr/core/src/table_function/user_defined.rs:200]: What about adding a link to the issue/PR?

File: src/expr/core/Cargo.toml [hunk 3]
```diff
@@ -22,7 +22,9 @@ embedded-python-udf = ["arrow-udf-python"]
 [dependencies]
 anyhow = "1"
 arrow-array = { workspace = true }
+arrow-flight = "50"
```
Comment (BugenZhao) [inline src/expr/core/Cargo.toml:25]: Can `arrow-udf-flight` re-export this?

**Review by BugenZhao — APPROVED** [review]
LGTM!

File: e2e_test/udf/test.py (hunk 1 above)
Comment (wangrunji0408) [reply e2e_test/udf/test.py:93]: Yes, it is a breaking change, but only for the new version. Existing users will not be affected unless they try to upgrade to the new version.

**Pull request merged**